x/sequencer/keeper/msg_server_decrease_bond.go

```diff
@@ -0,0 +1,37 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension/v3/x/sequencer/types"
+)
+
+// DecreaseBond implements types.MsgServer.
+func (k msgServer) DecreaseBond(goCtx context.Context, msg *types.MsgDecreaseBond) (*types.MsgDecreaseBondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sequencer, err := k.bondUpdateAllowed(ctx, msg.GetCreator())
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the bond reduction will make the sequencer's bond less than the minimum bond value
+	minBondValue := k.GetParams(ctx).MinBond
+	if !minBondValue.IsNil() && !minBondValue.IsZero() {
+		decreasedBondValue := sequencer.Tokens.Sub(msg.DecreaseAmount)
+		if decreasedBondValue.IsAllLT(sdk.NewCoins(minBondValue)) {
+			return nil, types.ErrInsufficientBond
+		}
+	}
+	completionTime := ctx.BlockHeader().Time.Add(k.UnbondingTime(ctx))
+	k.setDecreasingBondQueue(ctx, types.BondReduction{
```

Comment:
Resolved in https://github.com/dymensionxyz/dymension/pull/1031/commits/16251de313edec1048311a78e29104ae0d1b0671 by only allowing one bond reduction at a time

Reply:
maybe it's better to allow modifying the pending `Decrease` instead of waiting for it to complete.
@omritoptix wdyt

Reply:
I think every bond decrease should have it's own period (like in tenermint), e.g (you start with 100 bond):
1. you decrease bond 10 bond (it will take them 21 days)
2. after 2 days you decrease 5 more bond (it will also take them 21 days from the day u initiated it - i.e their own timeline).

I believe this was discussed [here in the ADR](https://www.notion.so/ADR-x-Sequencer-Bond-Management-b0a7061435fb40a28bd3cf64b899e968?d=0f92c1851c974c09ac4e86414d85d49c&pvs=4#8faf0ba648c2441298349f5cbb798f47).

This is good for predictability (so if a sequencer unbond 1% and than in the last day unbond 80% more the user doesn't have enough time to react to this change)

Reply:
@mtsitrin Made the modifications in https://github.com/dymensionxyz/dymension/pull/1031/commits/b622f6bab179d44b374c224a3f219689aa525043 to handle all the decrease bondings as a queue